components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java

```diff
@@ -0,0 +1,214 @@
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.apimgt.impl.APIManagerConfiguration;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class OrganizationPurgeDAO {
+
+    private static final Log log = LogFactory.getLog(ApiMgtDAO.class);
+    private static OrganizationPurgeDAO INSTANCE = null;
+    private boolean multiGroupAppSharingEnabled = false;
+
+    public OrganizationPurgeDAO() {
+        APIManagerConfiguration configuration = ServiceReferenceHolder.getInstance()
+                .getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        multiGroupAppSharingEnabled = APIUtil.isMultiGroupAppSharingEnabled();
+    }
+
+    /**
+     * Method to get the instance of the ApiMgtDAO.
+     *
+     * @return {@link OrganizationPurgeDAO} instance
+     */
+    public static OrganizationPurgeDAO getInstance() {
+
+        if (INSTANCE == null) {
+            INSTANCE = new OrganizationPurgeDAO();
+        }
+
+        return INSTANCE;
+    }
+
+    /**
+     * Get API data for given organization
+     *
+     * @param orgId organization Id
+     * @return ArrayList<APIIdentifier>
+     * @throws APIManagementException
+     */
+    public ArrayList<APIIdentifier> getAPIIdList(String orgId) throws APIManagementException {
+
+        ArrayList<APIIdentifier> apiList = new ArrayList<>();
+        try (Connection conn = APIMgtDBUtil.getConnection();
+                PreparedStatement ps = conn.prepareStatement(OrganizationPurgeSQLConstants.GET_API_LIST_SQL_BY_ORG)) {
+            ps.setString(1, orgId);
+            try (ResultSet rs = ps.executeQuery()) {
+                while (rs.next()) {
+                    int apiId = rs.getInt("API_ID");
+                    String apiUuid = rs.getString("API_UUID");
+                    String apiName = rs.getString("API_Name");
+                    String apiProvider = rs.getString("API_Provider");
+                    String apiVersion = rs.getString("API_Version");
+
+                    APIIdentifier apiIdentifier = new APIIdentifier(apiProvider, apiName, apiVersion, apiUuid);
+                    apiIdentifier.setId(apiId);
+                    apiList.add(apiIdentifier);
+                }
+            }
+        } catch (SQLException e) {
+            log.error("Error while getting apiUuid list of organization" + orgId, e);
+            handleException("Failed to get API apiUuid list of organization " + orgId, e);
+        }
+        return apiList;
+    }
+
+    /**
+     * Delete all organization API data
```

Comment:
make sure to add a `.` at the end of first sentense in javadoc description. this causes checkstyle failures